src/illuminatio/host.py

```diff
@@ -136,31 +150,30 @@ def __eq__(self, other):
         if isinstance(other, GenericClusterHost):
             return (self.namespace_labels == other.namespace_labels
                     and self.pod_labels == other.pod_labels)
-        else:
-            return False
+        return False
 
     def __hash__(self):
         return hash(str(self))
 
     def matches(self, obj):
         if obj is None:
             raise ValueError("obj to match to cannot be None")
-        elif isinstance(obj, k8s.client.V1Namespace):
+        if isinstance(obj, k8s.client.V1Namespace):
             return (obj.metadata.labels is not None
                     and all(item in obj.metadata.labels.items() for item in self.namespace_labels.items()))
-        else:
-            # we need to request the namespace from the cluster to match the labels TODO: find better solution
-            namespace = \
-                k8s.client.CoreV1Api().read_namespace(obj.metadata.namespace)
-            namespace_matches = (namespace.metadata.labels is not None and
-                                 all(item in namespace.metadata.labels.items() for item in
-                                     self.namespace_labels.items()))
-            if isinstance(obj, k8s.client.V1Pod):
-                return (namespace_matches and obj.metadata.labels is not None
-                        and all(item in obj.metadata.labels.items() for item in self.pod_labels.items()))
-            elif isinstance(obj, k8s.client.V1Service):
-                return (namespace_matches and obj.spec.selector is not None
-                        and all(item in obj.spec.selector.items() for item in self.pod_labels.items()))
+        # we need to request the namespace from the cluster to match the labels TODO: find better solution
+        namespace = \
+            k8s.client.CoreV1Api().list_namespace(field_selector="metadata.name=" + obj.metadata.namespace).items[0]
```

Comment:
This should be a `read_namespace ` and not a `list_namespace ` with a `field_selector` matching the exact name .